src/libsync/propagateuploadng.cpp

```diff
@@ -332,7 +494,54 @@ void PropagateUploadFileNG::startNextChunk()
     connect(job, &QObject::destroyed, this, &PropagateUploadFileCommon::slotJobDestroyed);
     job->start();
     propagator()->_activeJobList.append(this);
-    _currentChunk++;
+    updateRanges(_currentChunk, _currentChunkSize);
+}
+
+void PropagateUploadFileNG::slotZsyncGenerationFinished(QString fileName)
+{
+    qCDebug(lcPropagateUpload) << "Finished generation of:" << fileName << "size:" << FileSystem::getSize(fileName);
+
+    auto device = new UploadDevice(&propagator()->_bandwidthManager);
+
+    _fileSize += FileSystem::getSize(fileName);
+
+    if (!device->prepareAndOpen(fileName, 0, FileSystem::getSize(fileName))) {
+        qCWarning(lcPropagateUpload) << "Could not prepare upload device: " << device->errorString();
+
+        // If the file is currently locked, we want to retry the sync
+        // when it becomes available again.
+        if (FileSystem::isFileLocked(fileName)) {
+            emit propagator()->seenLockedFile(fileName);
+        }
```

Comment:
Minor: Probably unnecessary since this is not a user file

Reply:
Removed.